src/rgw/rgw_lc.cc

```diff
@@ -561,38 +561,30 @@ static int remove_expired_obj(
   del_op->params.unmod_since = meta.mtime;
   del_op->params.marker_version_id = version_id;
 
-  rgw::sal::RadosStore *rados = dynamic_cast<rgw::sal::RadosStore*>(oc.store);
-  if (rados) {
-    // notification supported only for RADOS store for now
-    notify
-      = store->get_notification(dpp, obj.get(), nullptr, event_type,
-			      bucket.get(), lc_id,
-			      const_cast<std::string&>(oc.bucket->get_tenant()),
-			      lc_req_id, null_yield);
-
-    /* can eliminate cast when reservation is lifted into Notification */
-    auto notify_res = static_cast<rgw::sal::RadosNotification*>(notify.get())->get_reservation();
-
-    ret = rgw::notify::publish_reserve(dpp, event_type, notify_res, nullptr);
-    if ( ret < 0) {
-      ldpp_dout(dpp, 1)
-        << "ERROR: notify reservation failed, deferring delete of object k="
-        << o.key
-        << dendl;
-      return ret;
-    }
+  // notification supported only for RADOS store for now
+  notify = store->get_notification(dpp, obj.get(), nullptr, event_type,
```

Comment:
so, we are just relying on getting nothing if !rados here?

Reply:
No, we get a Notification object, but it doesn't do anything.  DBNotification, for example, is a stub.  So it's functionally the same as skipping notifications when !rados